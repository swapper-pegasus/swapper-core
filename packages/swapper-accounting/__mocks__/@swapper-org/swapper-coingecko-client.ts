import {
  ResponseGetTokenPrice,
  ResponseGetTokens
} from '@swapper-org/swapper-coingecko-client'
export class SwapperCoingeckoClient {
  public getTokensPriceInUsd (): Promise<ResponseGetTokenPrice> {
    const response: ResponseGetTokenPrice = { bitcoin: { usd: '100000' }, ethereum: { usd: '1000' } }
    return new Promise((resolve) => resolve(response))
  }

  public getTokenDataBySymbol (): Promise<ResponseGetTokens> {
    const response: ResponseGetTokens = [{ id: 'ethereum', name: 'Ethereum', symbol: 'eth' }, { id: 'bitcoin', name: 'Bitcoin', symbol: 'btc' }]
    return new Promise((resolve) => resolve(response))
  }
}
