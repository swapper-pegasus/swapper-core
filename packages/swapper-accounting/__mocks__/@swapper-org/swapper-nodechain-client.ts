import {
  ResponseGetAddressesBalance
} from '@swapper-org/swapper-nodechain-client'

export class SwapperNodechainEthClient {
  public getAddressesBalance (): Promise<ResponseGetAddressesBalance> {
    const response: ResponseGetAddressesBalance = {
      id: 1,
      jsonrpc: 'getAddressesBalance',
      result: [{
        address: '0xABC',
        balance: {
          confirmed: '0xA',
          unconfirmed: '0x0'
        }
      }]
    }
    return new Promise((resolve) => resolve(response))
  }
}

export class SwapperNodechainBtcClient {
  public getAddressesBalance (): Promise<ResponseGetAddressesBalance> {
    const response: ResponseGetAddressesBalance = {
      id: 1,
      jsonrpc: 'getAddressesBalance',
      result: [{
        address: 'abc',
        balance: {
          confirmed: '0x1',
          unconfirmed: '0x0'
        }
      }]
    }
    return new Promise((resolve) => resolve(response))
  }
}
