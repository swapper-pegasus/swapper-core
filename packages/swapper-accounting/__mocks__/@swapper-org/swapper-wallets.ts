import {
  WalletBalance
} from '@swapper-org/swapper-wallets'
class EthWallet {
  public getBalance (): Promise<WalletBalance> {
    const response: WalletBalance = { confirmed: '0x1', unconfirmed: '0x0' }
    return new Promise((resolve) => resolve(response))
  }
}

export { EthWallet }
