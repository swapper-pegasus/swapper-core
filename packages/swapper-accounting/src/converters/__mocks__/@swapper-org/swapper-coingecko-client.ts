export class SwapperCoingeckoClient {
  public getTokensPriceInUsd = jest.fn().mockResolvedValue({})
  public getTokenDataBySymbol = jest.fn().mockResolvedValue({})
}
