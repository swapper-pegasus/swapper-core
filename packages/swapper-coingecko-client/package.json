{
  "name": "@swapper-org/swapper-coingecko-client",
  "version": "0.0.36",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "license": "(GPL-3.0-or-later OR AGPL-3.0-or-later)",
  "files": [
    "lib",
    "src"
  ],
  "scripts": {
    "build": "yarn build:cjs",
    "build:dev": "yarn tsc -b",
    "build:cjs": "yarn tsc --project ./tsconfig.build.json",
    "test": "jest --coverage"
  },
  "repository": {
    "type": "git",
    "url": "ssh://git@github.com:swapper-org/swapper-core.git",
    "directory": "packages/swapper-coingecko-client"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  },
  "devDependencies": {
    "@types/jest": "^27.0.2",
    "jest": "^27.3.1",
    "ts-jest": "^27.0.7",
    "typescript": "^4.4.3"
  },
  "devEngines": {
    "node": ">=14.x"
  },
  "dependencies": {
    "axios": "^0.24.0",
    "qs": "^6.10.1"
  }
}
