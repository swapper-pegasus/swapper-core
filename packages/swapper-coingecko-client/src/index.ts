import SwapperCoingeckoClient from './clients/SwapperCoingeckoClient'

export * from './clients/types'

export {
  SwapperCoingeckoClient
}
