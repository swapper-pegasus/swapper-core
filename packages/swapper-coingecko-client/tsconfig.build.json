{
  "extends": "../../tsconfig.base.json",
  "include": ["src"],
  "exclude": [
    "packages/*/src/__tests__",
    "packages/*/node_modules", 
    "packages/*/lib",
    "packages/*/coverage",
    "packages/*/dist",
    "packages/*/*.spec.ts",
    "packages/*/*tests"
  ],
}