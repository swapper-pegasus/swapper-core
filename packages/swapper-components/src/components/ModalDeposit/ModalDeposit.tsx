import React from 'react'
import { Modal, Button, Alert } from '@swapper-org/swapper-elements'
import { Token } from '@swapper-org/swapper-coingecko-client'

type Props = {
  address?: string,
  token?: Token,
  isOpen: boolean,
  onClose: () => void;
};

const DEFAULT_TOKEN = {
  id: '',
  name: '',
  symbol: ''
}

export function ModalDeposit ({ token = DEFAULT_TOKEN, address, isOpen, onClose }: Props) {
  return (
    <Modal isOpen={isOpen} onClose={() => onClose()}>
      <div data-testid='deposit-modal'>
        <div className="flex flex-col justify-center items-center">
          <Alert
            type='warning'
            description={`Sending any crypto asset besides ${token.name} to this address may result in the total loss of your funds`}
          />
          <div className="mt-3">
            <p className='font-medium'>{address}</p>
          </div>
        </div>
        <div className="bg-white pt-3 py-3 text-right">
          <Button dataTestId='modal-accept-button' onClick={() => onClose()}>
            <span>Continue</span>
          </Button>
        </div>
      </div>
    </Modal>
  )
}

ModalDeposit.defaultProps = {
  token: DEFAULT_TOKEN
}
