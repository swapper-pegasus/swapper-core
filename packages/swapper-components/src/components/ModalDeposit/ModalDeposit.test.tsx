import React, { useState } from 'react'
import '@testing-library/jest-dom'
import { ModalDeposit } from './ModalDeposit'
import { Button } from '@swapper-org/swapper-elements'
import { render, fireEvent } from '@testing-library/react'
import { Token } from '@swapper-org/swapper-coingecko-client'

type Props = {
    address: string,
    token?: Token,
    onClose: () => void
}

const ModalTestComponent = (props: Props) => {
  const [isOpen, setIsOpen] = useState(false)
  return (
        <>
            <div data-testid='modal-container' id='modalContainer'></div>
            <Button onClick={() => setIsOpen(true)} data-testid='button-open-modal'></Button>
            <ModalDeposit {...props} isOpen={isOpen} />
        </>
  )
}

const defaultProps = {
  address: '0xABC',
  isOpen: true,
  onClose: jest.fn()
}

function setUpRender (props: object) {
  const finalProps = {
    ...defaultProps,
    ...props
  }
  return render(
    <ModalTestComponent {...finalProps}/>
  )
}

describe('ModalDeposit', () => {
  it('Render with custom token', () => {
    const component = setUpRender({ token: { name: 'Bitcoin', symbol: 'btc', id: '1' } })
    fireEvent.click(component.getByRole('button'))
    expect(component.baseElement).toMatchSnapshot()
  })
  it('Render with default token', () => {
    const component = setUpRender({})
    fireEvent.click(component.getByRole('button'))
    expect(component.baseElement).toMatchSnapshot()
  })
  it('on close with keyboard', () => {
    const mockOnClose = jest.fn()
    const component = setUpRender({ onClose: mockOnClose })
    fireEvent.click(component.getByRole('button'))
    fireEvent.keyDown(component.getByRole('dialog'), { key: 'esc', keyCode: 27 })
    expect(mockOnClose).toHaveBeenCalledTimes(1)
  })
  it('on close with button', () => {
    const mockOnClose = jest.fn()
    const component = setUpRender({ onClose: mockOnClose })
    fireEvent.click(component.getByRole('button'))
    fireEvent.click(component.getByTestId('modal-accept-button'))
    expect(mockOnClose).toHaveBeenCalledTimes(1)
  })
})
