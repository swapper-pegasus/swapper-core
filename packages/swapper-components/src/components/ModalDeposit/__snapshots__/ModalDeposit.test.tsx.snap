// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalInfo Render with description 1`] = `
<body>
  <div>
    <div
      data-testid="modal-container"
      id="modalContainer"
    >
      <div
        aria-labelledby="modal-title"
        aria-modal="true"
        class="fixed inset-0 overflow-y-auto w-3/6 m-auto"
        role="dialog"
      >
        <div
          class="flex items-center justify-center min-h-screen pt-4 px-4 pb-20 text-center"
        >
          <div
            aria-hidden="true"
            class="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity"
          />
          <span
            aria-hidden="true"
            class="hidden"
          >
            â€‹
          </span>
          <div
            class="inline-block align-bottom bg-white rounded-lg text-left overflow-hidden shadow-xl transform transition-all p-6"
          >
            <div
              class="bg-white"
            >
              <div
                data-testid="deposit-modal"
              >
                <div
                  class="flex flex-col justify-center items-center"
                >
                  <div
                    class="flex items-center bg-warning border-l-4 border-warning-darkest text-warning-dark py-2 min-w-full"
                  >
                    <div
                      class="px-2"
                    >
                      <svg
                        aria-hidden="true"
                        class="w-6 h-6"
                        fill="none"
                        stroke="currentColor"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                        />
                      </svg>
                    </div>
                    <div>
                      <p
                        class="font-semibold"
                      />
                      <p>
                        Sending any crypto asset besides Bitcoin to this address may result in the total loss of your funds
                      </p>
                    </div>
                  </div>
                  <div
                    class="mt-3"
                  >
                    <p
                      class="font-medium"
                    >
                      0xABC
                    </p>
                  </div>
                </div>
                <div
                  class="bg-white pt-3 py-3 text-right"
                >
                  <button
                    class="disabled:cursor-not-allowed disabled:opacity-20 bg-primary rounded-full hover:bg-primary-dark text-white text-sm px-8 py-2"
                    data-testid="modal-accept-button"
                  >
                    <span>
                      Continue
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="disabled:cursor-not-allowed disabled:opacity-20 bg-primary rounded-full hover:bg-primary-dark text-white text-sm px-8 py-2"
    />
  </div>
</body>
`;
